shell/platform/darwin/common/framework/Source/FlutterChannelsTest.m

```diff
@@ -0,0 +1,143 @@
+                                 FlutterBinaryReply reply = obj;
+                                 reply(nil);
+                                 return YES;
+                               }]]);
+  [channel invokeMethod:@"foo"
+              arguments:@[ @(1) ]
+                 result:^(id _Nullable result) {
+                   didCallReply = YES;
+                   XCTAssertEqual(FlutterMethodNotImplemented, result);
+                 }];
+  OCMVerify(binaryMessenger);
+  XCTAssertTrue(didCallReply);
+}
+
+- (void)testMethodMessageHandler {
+  NSString* channelName = @"foo";
+  id binaryMessenger = OCMProtocolMock(@protocol(FlutterBinaryMessenger));
+  id codec = OCMProtocolMock(@protocol(FlutterMethodCodec));
+  FlutterMethodChannel* channel = [[FlutterMethodChannel alloc] initWithName:channelName
+                                                             binaryMessenger:binaryMessenger
+                                                                       codec:codec];
+  XCTAssertNotNil(channel);
+
+  NSData* encodedMethodCall = [@"hey" dataUsingEncoding:NSUTF8StringEncoding];
+  OCMStub([codec encodeMethodCall:[OCMArg any]]).andReturn(encodedMethodCall);
+  FlutterMethodCallHandler handler =
+      ^(FlutterMethodCall* _Nonnull call, FlutterResult _Nonnull result) {
+        NSLog(@"hey");
+      };
+  [channel setMethodCallHandler:handler];
+  OCMVerify([binaryMessenger setMessageHandlerOnChannel:channelName
+                                   binaryMessageHandler:[OCMArg isNotNil]]);
+}
+
+- (void)testCallMethodHandler {
+  NSString* channelName = @"foo";
+  MockBinaryMessenger* binaryMessenger = [[MockBinaryMessenger alloc] init];
+  id codec = OCMProtocolMock(@protocol(FlutterMethodCodec));
+  FlutterMethodChannel* channel = [[FlutterMethodChannel alloc] initWithName:channelName
+                                                             binaryMessenger:binaryMessenger
+                                                                       codec:codec];
+  XCTAssertNotNil(channel);
+
+  NSData* encodedMethodCall = [@"encoded" dataUsingEncoding:NSUTF8StringEncoding];
+  NSData* replyData = [@"reply" dataUsingEncoding:NSUTF8StringEncoding];
+  NSData* replyEnvelopeData = [@"reply-envelope" dataUsingEncoding:NSUTF8StringEncoding];
+  FlutterMethodCall* methodCall = [[FlutterMethodCall alloc] init];
+  OCMStub([codec decodeMethodCall:encodedMethodCall]).andReturn(methodCall);
+  OCMStub([codec encodeSuccessEnvelope:replyData]).andReturn(replyEnvelopeData);
+  __block BOOL didCallHandler = NO;
+  __block BOOL didCallReply = NO;
+  FlutterMethodCallHandler handler =
+      ^(FlutterMethodCall* _Nonnull call, FlutterResult _Nonnull result) {
+        XCTAssertEqual(methodCall, call);
+        didCallHandler = YES;
+        result(replyData);
+      };
+  [channel setMethodCallHandler:handler];
+  binaryMessenger.handlers[channelName](encodedMethodCall, ^(NSData* reply) {
```

Comment:
Do you need a real MockBinaryMessenger class?  Can you verify the reply data as it happens instead with an `adDo` and pull the reply off the block parameter?

Reply:
No, I tried to implement it with OCMock at first but failed so I switched to a real Mock object.  I think now that i've played around with it a bit more I could get it working with OCMock, it didn't seem worth the effort at this point considering it's a test.